contrib/ucx-dev.in

```diff
@@ -0,0 +1,31 @@
+#%Module1.0#####################################################################
+##
+## Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+##
+## See file LICENSE for terms.
+
+##
+## modulefiles/ucx-dev
+##
+## Usage:
+## - module load ./contrib/ucx-dev
+## - module unload ucx-dev
+
+proc ModulesHelp { } {
+        puts stderr "\tThis is ucx-dev\n"
+        puts stderr "\tUsed for development only\n"
+}
+
+set     root            "@prefix@"
+set     module_prefix   $root
+
+module-whatis   "ucx-dev"
+prepend-path    PATH                    $root/bin
+prepend-path    LD_LIBRARY_PATH         $root/lib
+prepend-path    LIBRARY_PATH            $root/lib
+prepend-path    C_INCLUDE_PATH          $root/include
+prepend-path    CPLUS_INCLUDE_PATH      $root/include
+prepend-path    INCLUDE                 $root/include
+prepend-path    CPATH                   $root/include
+prepend-path    FPATH                   $root/include
```

Comment:
maybe point to build directory, so user will not have to "make install" before using the module?

Reply:
Since make install does some directory adding/shuffing/libtool, i'd rather not, to make it easier to maintain, making sure that when module loading over say ucx-1.14, we can't end-up using 1.14 header/lib.

Also I don't think users of say `#include <ucs/type/status.h>` would be able to build if we refer to include directory directly in the src directories.

Reply:
so maybe rename the module from ucx-dev to ucx?
since it points to "installed" paths and not "development" paths

Reply:
fixed